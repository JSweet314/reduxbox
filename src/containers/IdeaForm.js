import React, {Component} from 'react';

export default class IdeaForm extends Component {
  constructor() {
    super();
    this.state = {
      titleInput: '',
      bodyInput: '',
      isDisabled: true
    };
  }

  updateTitleInput(event) {
    const waitingForUserInput = event.target.value && this.state.bodyInput;

    this.setState({
      titleInput: event.target.value,
      isDisabled: !waitingForUserInput
    });
  }

  updateBodyInput(event) {
    const waitingForUserInput = event.target.value && this.state.titleInput;

    this.setState({
      bodyInput: event.target.value,
      isDisabled: !waitingForUserInput
    });
  }

  createIdea() {
    const idea = {
      title: this.state.titleInput,
      body: this.state.bodyInput,
      id: Date.now(),
      quality: 'swill'
    };

    // dispatch(addIdea(idea));
  }

  render() {
    return (
      <form>
        <h1>reduxbox</h1>
        <input 
          className="title-input" 
          type="text"
          value={this.state.titleInput}
          onChange={event => this.updateTitleInput(event)}
        />
        <input 
          className="body-input" 
          type="text" 
          value={this.state.bodyInput}
          onChange={event => this.updateBodyInput(event)}
        />
        <button 
          className="saveBtn"
          disabled={this.state.isDisabled}
          onClick={() => createIdea()}
        >
          save
        </button>
      </form>
    );
  }
}