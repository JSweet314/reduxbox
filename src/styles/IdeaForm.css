form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #e5f3f2;
  color: #6d6e71;
  font-family: 'Open Sans', sans-serif;
  width: 100%;
  text-align: center;
}

h1 {
  color: #6d6e71;
  font-family: 'Roboto Slab', serif;
  font-size: 4rem;
}

h1 span {
  color: #00a79d;
}

form input {
  line-height: 1;
  padding: 0 0 0 0.5rem;
}

form input,
form button {
  width: 50%;
  height: 4rem;
  border: 2px solid lightgray;
  font-size: 2rem;
  font-family: 'Open Sans', sans-serif;
  margin: 0;
}

form button {
  background-color: #00a79d;
  color: white;
  width: 50%;
  border: none;
}

form button:enabled:hover {
  cursor: pointer;
  background-color: #045e55;
}

form p {
  text-align: right;
  width: 50%;
  margin: 0.5rem 0 1rem 0;
  font-size: 1.2rem;
}