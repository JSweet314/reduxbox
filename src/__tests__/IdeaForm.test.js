import React from 'react';
import {shallow, mount} from 'enzyme';
import IdeaForm from '../containers/IdeaForm';

describe('IdeaForm', () => {
  let ideaForm;

  beforeEach(() => {
    ideaForm = shallow(<IdeaForm />);
  });

  it('should render a form', () => {
    expect(ideaForm.find('form')).toBeDefined();
  });

  it('should contain an h1 tag with the text reduxbox', () => {
    const expected = <h1>reduxbox</h1>;
    
    expect(ideaForm.contains(expected)).toEqual(true);
  });

  it('should contain two input fields for title and body', () => {
    expect(ideaForm.find('input.title-input').length).toEqual(1);
    expect(ideaForm.find('input.body-input').length).toEqual(1);
  });

  it('should contain a save button', () => {
    expect(ideaForm.find('.saveBtn').length).toEqual(1);
  });
});
