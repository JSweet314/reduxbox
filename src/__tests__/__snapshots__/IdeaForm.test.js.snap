// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaForm should render a form 1`] = `
<form
  action=""
>
  <h1>
    reduxbox
  </h1>
</form>
`;
